jsonrpc/codec.go

```diff
@@ -102,6 +103,78 @@ const (
 
 type BlockNumber int64
 
+type BlockNumberOrHash struct {
+	BlockNumber *BlockNumber `json:"blockNumber,omitempty"`
+	BlockHash   *types.Hash  `json:"blockHash,omitempty"`
+}
+
+// Unmarshal will try to extract the filter's data.
+// Here are the possible input formats :
+//
+// 1 - "latest", "pending" or "earliest"	- self-explaining keywords
+// 2 - "0x2"								- block number #2 (EIP-1898 backward compatible)
+// 3 - {blockNumber:	"0x2"}				- EIP-1898 compliant block number #2
+// 4 - {blockHash:		"0xe0e..."}			- EIP-1898 compliant block hash 0xe0e...
+func (bnh *BlockNumberOrHash) Unmarshal(input *interface{}) error {
```

Comment:
Please add a unit test to test this.

Reply:
Added in https://github.com/0xPolygon/polygon-sdk/pull/284/commits/14c3e134c8727de6332cda429dff4f1e8527a0c6